frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.component.ts

```diff
@@ -147,6 +164,41 @@ export class ModelDiagramDialogComponent {
         saveAs(response, `${uuid}.svg`);
       });
   }
+
+  async getUser() {
+    const token = localStorage.getItem('access_token');
+
+    const headers = this.getHeaders(token);
+
+    const response = await fetch(environment.backend_url + '/users/current', {
+      headers,
+    });
+    return response.json();
+  }
+
+  async insertToken() {
+    const user = await this.getUser();
+    const id = user.id;
+    const token = localStorage.getItem('access_token');
+    this.isButtonClicked = true;
+    const headers = this.getHeaders(token);
+
+    const response = await fetch(
+      environment.backend_url + `/users/${id}/token`,
+      {
+        headers,
+      }
+    );
+    const basic_token = await response.text();
+    this.passwordValue = basic_token.replaceAll('"', '');
+  }
+
+  getHeaders(token: string | null) {
+    return {
+      Authorization: `Bearer ${token}`,
+      'Content-Type': 'application/json',
+    };
+  }
```

Comment:
We have an auth interceptor that automatically adds the authorisation header for each outgoing request (in the `injectAccessToken` method). Therefore, we do not have to create the headers by outselves.